citesphere/src/main/java/edu/asu/diging/citesphere/web/user/concepts/EditConceptController.java

```diff
@@ -59,6 +57,10 @@ public String post(Model model, @PathVariable("conceptId") String conceptId, Aut
             redirectAttributes.addFlashAttribute("show_alert", true);
             redirectAttributes.addFlashAttribute("alert_msg", "Concept was successfully saved.");
             redirectAttributes.addFlashAttribute("alert_type", "success");
+        } else {
+            model.addAttribute("form", form);
+            result.rejectValue("uri", "uri", "A concept with this uri exists.");
+            return "auth/concepts/edit";
```

Comment:
this kind of validation should be done in the validator

Reply:
also, where is the validator?

Reply:
@jdamerow I have added the validation annotations in the next PR. 
Do you suggest that I add a validator class to check if URI exists? This would require me to autowire User and Concept manager.

Reply:
what do you mean by "added the validation annotations in the next PR". Do you mean you've added it and it will show up when you reopen the PR? 
And the @Valid only does something if you have a validator class that actually does the validation, or if you use and create validation annotations for the classes themselves.
And yes, the validator should make sure the data is valid. So if the user already has a concept with the URI the validation should fail. There should be no more validation in the controller itself (except where not possible otherwise).